Modules/RatMama.py

```diff
@@ -0,0 +1,206 @@
+    \s+-\s+                              #  -
+    System:\s*(?P<system>.*?)            # Match system name
+    (?:\s[sS][yY][sS][tT][eE][mM]|)      # Strip " system" from end, if present (case insensitive)
+    \s+-\s+                              #  -
+    Platform:\s*(?P<platform>\w+)        # Match platform (currently can't contain spaces)
+    \s+-\s+                              #  -
+    O2:\s*(?P<o2>.+?)                    # Match oxygen status
+    \s+-\s+                              #  -
+    Language:\s*
+    (?P<full_language>                   # Match full language text (for regenerating the line)
+    (?P<language>.+?)\s*                 # Match language name. (currently unused)
+    \(                                   # The "(" of "(en-US)"
+    (?P<language_code>.+?)               # "en"
+    (?:                                  # Optional group
+        -(?P<language_country>.+)        # "-", "US" (currently unused)
+    )?                                   # Actually make the group optional.
+    \)                                   # The ")" of "(en-US)"
+    )                                    # End of full language text
+    (?:                                  # Possibly match IRC nickname
+    \s+-\s+                              #  -
+    IRC\s+Nickname:\s*(?P<nick>[^\s]+)   # IRC nickname
+    )?                                   # ... emphasis on "Possibly"
+    )                                    # End of the main capture group
+    \s*                                  # Handle any possible trailing whitespace
+    $                                    # End of pattern
+""")
+
+
+@rule(r"^\s*Incoming Client:", case_sensitive=False, full_message=False, prefixless=True,
+      pass_match=False)
+async def handle_ratmama_announcement(ctx: Context) -> None:
+
+    """
+    Handles the Announcement made by RatMama.
+    Details are extracted, wrapped in a Rescue object and appended to the Rescue board.
+    An appropriate answer will be sent to IRC.
+
+    Args:
+        ctx: Context of the announcement
+
+    Returns: None
+
+    """
+
+    if ctx.user.nickname.casefold() not in (
+            k.casefold() for k in config.config["ratsignal_parser"]["announcer_nicks"]
+    ):
+        return
+
+    message: str = ctx.words_eol[0]
+    result = re.fullmatch(ratmama_regex, message)
+    client_name: str = result.group("cmdr")
+    system_name: str = result.group("system")
+    platform_name: str = result.group("platform")
+    o2_status: bool = result.group("o2") == "OK"  # false is CR
+    lang_code: str = result.group("language_code")
+    nickname: Optional[str] = result.group("nick")
+
+    exist_rescue: Rescue = ctx.bot.board.find_by_name(client_name)
+
+    if exist_rescue:
+        # we got a case already!
+        await ctx.reply(f"{client_name} has reconnected! Case #{exist_rescue.board_index}")
+        # now let's make it more visible if stuff changed
+        diff_response: str = ""
```

Comment:
redundant type annotation.
```suggestion
        diff_response = ""
```